libs/pages/dashboard/src/lib/feature/tasks-grouped-by-status/tasks-grouped-by-status.component.ts

```diff
@@ -0,0 +1,192 @@
+  public removedItems: Item[] = [];
+  public groups: Group[] = [];
+
+  public total: number | null = null;
+
+  public loadTasks$ = timer(0, 2000).pipe(
+    switchMap(() => this._grpcTasksService.countTasksByStatus$()),
+    map((data) => {
+      return data.status ?? [];
+    }),
+    tap((status) => {
+      this.total = this._calculateTotal(status);
+    })
+  );
+
+  constructor(
+    private _grpcTasksService: GrpcTasksService,
+    private _dashboardTasksService: DashboardTasksService
+  ) {}
+
+  public ngOnInit(): void {
+    const { groups, removedItems } = this._dashboardTasksService.load();
+    this.groups = groups;
+    this.removedItems = removedItems;
+  }
+
+  /**
+   * This method is called when the user have moved an item. The behavior is handled by the component itself.
+   */
+  public onMoveItem(): void {
+    this._save();
+  }
+
+  /**
+   * This method is called when the user want to hide an item.
+   *
+   * @param itemId The id of the item to hide.
+   *
+   * @returns void
+   */
+  public onHideItem(itemId: Item['id']): void {
+    const item = this.groups.reduce((acc, group) => {
+      const item = group.items.find((item) => item.id === itemId);
+
+      if (item) {
+        group.items.splice(group.items.indexOf(item), 1);
+        acc = item;
+      }
+
+      return acc;
+    }, null as Item | null);
+
+    if (!item) {
+      return;
+    }
+
+    this.removedItems.push(item);
+
+    this._save();
+  }
+
+  /**
+   * This method is called when the user want to rename an existing group.
+   *
+   * @param data An object containing the old name and the new name.
+   *
+   * @returns void
+   */
+  public onRenameGroup({
+    oldName,
+    newName,
+  }: {
+    oldName: string;
+    newName: string;
+  }): void {
+    const group = this.groups.find((group) => group.name === oldName);
+
+    if (!group) {
+      return;
+    }
+
+    group.name = newName;
+
+    this._save();
+  }
+
+  /**
+   * This method is called when the user want to delete a group.
+   *
+   * @param groupName The name of the group to delete.
+   *
+   * @returns void
+   */
+  public onDeleteGroup(groupName: string): void {
+    const group = this.groups.find((group) => group.name === groupName);
+
+    if (!group) {
+      return;
+    }
+
+    this.removedItems.push(...group.items);
+    this.groups.splice(this.groups.indexOf(group), 1);
+
+    this._save();
+  }
+
+  /**
+   * This method is called when the user want to map a removed item to a group.
+   *
+   * @param pairing A map of item id and group.
+   *
+   * @returns void
+   */
+  public onUpdateRemovedItems(pairing: Map<Item['id'], Group>): void {
+    this.removedItems.forEach((item) => {
+      const group = pairing.get(item.id);
+
+      console.log(group);
+
+      if (!group) {
+        return;
+      }
+
+      group.items.push(item);
+      this.removedItems.splice(this.removedItems.indexOf(item), 1);
+    });
+
+    this._save();
+  }
```

Comment:
When having a list of items, the last element was not added to a group. This fix the bug.

```suggestion
  /**
   * This method is called when the user want to map a removed item to a group.
   *
   * @param pairing A map of item id and group.
   *
   * @returns void
   */
  public onUpdateRemovedItems(pairing: Map<Item['id'], Group>): void {
    const addedItems: Item[] = [];

    this.removedItems.forEach((item) => {
      const group = pairing.get(item.id);

      if (!group) {
        return;
      }

      group.items.push(item);
      addedItems.push(item)
      
    });
    
    // Remove all added Items
    addedItems.forEach((item: Item) => {
      this.removedItems.splice(this.removedItems.indexOf(item), 1);
    })

    this._save();
  }
```

Reply:
Who, nice catch!

Reply:
fixed